docs/standard/native-interop/disabled-marshaling.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Disabled runtime marshalling - .NET
+description: Learn how .NET marshals your types to a native representation when runtime marshalling is disabled.
+ms.date: 01/12/2022
+---
+
+# Disabled runtime marshalling
+
+When the [`System.Runtime.CompilerServices.DisableRuntimeMarshallingAttribute`](xref:System.Runtime.CompilerServices.DisableRuntimeMarshallingAttribute) attribute is applied to an assembly, the runtime disables most built-in support for data marshalling between managed and native representations. This document describes the features that are disabled and how .NET types map to native types when marshalling is disabled.
+
+## Scenarios where marshalling is disabled
+
+When the `DisableRuntimeMarshallingAttribute` is applied to an assembly, it affects P/Invokes and Delegate types in the assembly, as well as any calls to unmanaged function pointers in the assembly. It does not affect any P/Invoke or interop delegate types defined in other assemblies. It also does not disable marshalling for the runtime's built-in COM interop support. The built-in COM interop support can be enabled or disabled through a [feature switch](https://github.com/dotnet/runtime/blob/main/docs/workflow/trimming/feature-switches.md).
+
+### Disabled features
+
+When the `DisableRuntimeMarshallingAttribute` is applied to an assembly, the following attributes will have no effect or throw an exception:
+
+- [`LCIDConversionAttribute`](xref:System.Runtime.InteropServices.LCIDConversionAttribute) on a P/Invoke or a delegate
+- `SetLastError=true` on a P/Invoke
+- `ThrowOnUnmappableChar=true` on a P/Invoke
+- `BestFitMapping=true` on a P/Invoke
+- .NET variadic argument method signatures (varargs)
+- `in`, `ref`, `out` parameters
+
+## Default rules for marshaling common types
```

Comment:
```suggestion
## Default rules for marshalling common types
```

Reply:
Looking in the redirection file, we renamed to single-l "marshaling" a while back in #12005. See https://github.com/dotnet/docs/issues/11986.

Reply:
Bah. So, what are we doing here? I thought we agreed on two ll, but could be misremembering.

Reply:
I think we as the interop team agreed on two Ls, but we never updated the style guide that the docs team uses.

@gewarren, can we update the style guide and rename things back to "marshalling"?

Reply:
I've submitted a request to update the style guide. Let's see where this goes.

Reply:
It would be good to also update and ensure its covered in https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions#capitalizing-compound-words-and-common-terms

CC. @bartonjs as he may know if we have concrete guidance on `marshaling` vs `marshalling`

Reply:
Based on email from Celia Parker today (she spent 7 hours researching this), it sounds like we are sticking to one `l` for now.

"[...] That should be reason enough to support the argument of not changing "marshal" to "marshall" in the IESG.
Unless you have much more compelling reasoning with more powerful evidence to support it, I believe we should leave the IESG as it is."

That said, is this good to merge?

Reply:
I think so, unless @bartonjs has any pushback based on the FDG

Reply:
-- (We can also always fix it up later if there is pushback there)

Reply:
It's not in the standard spellings table.

Based on existing public types (which @tannergooding already pointed out), I believe our de facto in .NET is `-ller` and `-lling`.  Here are all the hits of Marshal in our refs.cs files where it's not the last word in the type/method name or followed by a capital letter:

```
System.Reflection.Metadata.cs:        public System.Reflection.Metadata.BlobHandle GetMarshallingDescriptor() { throw null; }
System.Reflection.Metadata.cs:        public System.Reflection.Metadata.BlobHandle GetMarshallingDescriptor() { throw null; }
System.Reflection.Metadata.cs:        public void AddMarshallingDescriptor(System.Reflection.Metadata.EntityHandle parent, System.Reflection.Metadata.BlobHandle descriptor) { }
System.Runtime.InteropServices.cs:    public partial interface ICustomMarshaler
System.Runtime.InteropServices.cs:        [System.ObsoleteAttribute("Marshalling as Currency may be unavailable in future releases.")]
System.Runtime.InteropServices.cs:        [System.ObsoleteAttribute("Marshalling as VBByRefString may be unavailable in future releases.")]
System.Runtime.InteropServices.cs:        [System.ObsoleteAttribute("Marshalling as AnsiBStr may be unavailable in future releases.")]
System.Runtime.InteropServices.cs:        [System.ObsoleteAttribute("Marshalling as TBstr may be unavailable in future releases.")]
System.Runtime.InteropServices.cs:        [System.ObsoleteAttribute("Marshalling arbitrary types may be unavailable in future releases. Please specify the type you wish to marshal as.")]
System.Runtime.InteropServices.cs:        public static void RegisterForMarshalling(ComWrappers instance) { }
```

Counting the obsoletion messages, it's 9:1, ignoring them, it's 4:1.  (And that's not counting things like the `MarshallerHelpers` from the new interop generator)

Wikipedia's current style for the word marshalling is the two-l form (https://en.wikipedia.org/wiki/Marshalling_(computer_science)).  While I generally agree with "US spelling" when there's a country-based difference, in this case, I believe one-l is wrong.